packages/gdrcopy.spec

```diff
@@ -55,17 +52,31 @@ Kernel-mode driver for GDRCopy.
 
 %build
 echo "building"
-make -j CUDA=${CUDA}
+make -j CUDA=%{CUDA} config lib exes
 
 %install
+# Install gdrcopy library and tests
 make install DESTDIR=$RPM_BUILD_ROOT prefix=%{_prefix} libdir=%{_libdir}
-make drv_install DESTDIR=$RPM_BUILD_ROOT 
+
+# Install gdrdrv src
+mkdir -p $RPM_BUILD_ROOT/usr/src
+cp -r -a $RPM_BUILD_DIR/%{name}-%{version}/src/gdrdrv $RPM_BUILD_ROOT%{usr_src_dir}/gdrdrv-%{version}
+cp -a $RPM_BUILD_DIR/%{name}-%{version}/dkms.conf $RPM_BUILD_ROOT%{usr_src_dir}/gdrdrv-%{version}
 
 # Install gdrdrv service script
 install -d $RPM_BUILD_ROOT/etc/init.d
 install -m 0755 $RPM_BUILD_DIR/%{name}-%{version}/init.d/gdrcopy $RPM_BUILD_ROOT/etc/init.d
 
 %post %{kmod}
+dkms add -m gdrdrv -v %{version} -q || :
+
+# Rebuild and make available for the all installed kernel
+for kver in $(ls -1d /lib/modules/* | cut -d'/' -f4)
```

Comment:
that dir may contain other stuff:
```
$ ls -l /lib/modules/
total 12
drwxr-xr-x 3 root root   18 Jan 13  2020 3.10.0-1062.1.1.el7.x86_64/
drwxr-xr-x 7 root root 4096 Feb 10  2020 3.10.0-1062.12.1.el7.x86_64/
drwxr-xr-x 3 root root   18 Jan 13  2020 3.10.0-1062.4.1.el7.x86_64/
drwxr-xr-x 3 root root   18 Sep 25 01:52 3.10.0-1062.4.3.el7.x86_64/
drwxr-xr-x 7 root root 4096 Feb 10  2020 3.10.0-1062.9.1.el7.x86_64/
drwxr-xr-x 3 root root   18 Sep 17  2019 3.10.0-1062.el7.x86_64/
drwxr-xr-x 3 root root   18 Sep 17  2019 3.10.0-862.14.4.el7.x86_64/
drwxr-xr-x 2 root root 4096 Sep 17  2019 3.10.0-957.1.3.el7.x86_64/
lrwxrwxrwx 1 root root   11 Feb  7  2020 kabi-current -> kabi-rhel77/
drwxr-xr-x 2 root root  117 Feb  7  2020 kabi-rhel70/
drwxr-xr-x 2 root root  117 Feb  7  2020 kabi-rhel71/
drwxr-xr-x 2 root root  117 Feb  7  2020 kabi-rhel72/
drwxr-xr-x 2 root root  117 Feb  7  2020 kabi-rhel73/
drwxr-xr-x 2 root root  117 Feb  7  2020 kabi-rhel74/
drwxr-xr-x 2 root root  117 Feb  7  2020 kabi-rhel75/
drwxr-xr-x 2 root root  117 Feb  7  2020 kabi-rhel76/
drwxr-xr-x 2 root root  117 Feb  7  2020 kabi-rhel77/

```

Reply:
That's ok. The script will successfully only target that is ready (already tested). Even if that folder is a valid linux kernel, gdrdrv will not be installed on that kernel if it is missing the header or nvidia src.